src/types.ts

```diff
@@ -11,3 +11,15 @@ export type ResultOf<T> = {
 };
 
 export type Optional<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+/**
+ * This allow to define a type with at least one attribute of T.
+ */
+export type AnyOf<T> = {
+  [K in keyof T]: Pick<T, K>;
+}[keyof T];
+
+/**
+ * This allow to define a type with at least one attribute of T that is not a key in E.
+ */
+export type AnyOfExcept<T, E extends keyof T> = AnyOf<Omit<T, E>>;
```

Comment:
I see that this helps you to build the types of the patch and post payloads. 
Would it be clearer to have:
```ts
type ShortLink = {
  createdAt: string;
  alias: string;
  platform: ...;
  item: { id: string };
}

type ShortLinkPostPayload = Omit<ShortLink, 'createdAt'>
type ShortLinkUpdatePayload = Omit<ShortLink, 'createdAt' | 'item'>

// if you want to allow consumers to only specify "some" parts of the ShortLink for the update:
type ShortLinkPatchPayload = Pick<ShortLink, 'alias'> & Partial<Omit<ShortLink, 'createdAt' | 'item' | 'alias'>>
```

Reply:
I think that the type 

```
type ShortLinkPatchPayload = AnyOfExcept<ShortLinkPayload, 'item'>;
```

is more clear than 
```
type ShortLinkPatchPayload = Pick<ShortLink, 'alias'> & Partial<Omit<ShortLink, 'createdAt' | 'item' | 'alias'>>
```

The advantage of the AnyOf and AnyOfExcept types is that you can easily reuse them for other types, while retaining (in my opinion) a clear idea of what the new type will do.

Concerning the types:
```
type ShortLink = {
  createdAt: string;
  alias: string;
  platform: ...;
  item: { id: string };
}

type ShortLinkPostPayload = Omit<ShortLink, 'createdAt'>
```

I think it is more concise as my current:
```
export interface ShortLinkPayload {
  alias: string;
  platform: keyof typeof ShortLinkPlatform;
  item: { id: string };
}

export interface ShortLinkResponse extends ShortLinkPayload {
  createdAt: string;
}
```
So I could replace my interfaces by types. Honestly, I never really know when to use types instead of interfaces.